src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -167,15 +168,23 @@ trait EntityComponent {
       allEntitiesAction.flatMap(cloneEntities(destWorkspaceContext, _))
     }
 
+    def loadEntityId(workspaceId: UUID, entity: Entity): ReadAction[Option[Long]] = {
+      uniqueResult(entityQuery.findEntityByName(workspaceId, entity.entityType, entity.name).map(_.id))
+    }
+
     def cloneEntities(destWorkspaceContext: SlickWorkspaceContext, entities: TraversableOnce[Entity]): ReadWriteAction[Unit] = {
       //1. First save JUST the entity record for the entity without any attributes (references)
       val insertEntitiesOnly = DBIO.seq(entities.map(entity => save(destWorkspaceContext, entity.copy(attributes = Map.empty))).toSeq:_*)
```

Comment:
I would take this a step further and call (entityQuery returning entityQuery.map(_.id)) += EntityRecord(...) directly, capturing the mapping between the returned id and the Entity